src/clients/users_proposal.rb

```diff
@@ -76,134 +54,92 @@ def main
-            _("Password Encryption Method: %1"),
-            Ops.get(@encoding2label, UsersSimple.EncryptionMethod, "")
-          )
-        end
-        @ret = {
-          "preformatted_proposal" => @rest == "" ?
-            HTML.List([@prop, @root_proposal]) :
-            HTML.List([@prop, @root_proposal, @rest]),
+        # TRANSLATORS: summary line. Second %s is the name of the method
+        encrypt = _("Password Encryption Method: <a href=%s>%s</a>") %
+          ["users--encryption", ::Users::EncryptionMethod.current.label]
+        ret = {
+          "preformatted_proposal" => HTML.List([prop, root_proposal, encrypt]),
           "language_changed"      => false,
-          "links"                 => ["users--user", "users--root"]
+          "links"                 => ["users--user", "users--root", "users--encryption"]
         }
-      elsif @func == "Description"
-        @ret = {
+      elsif func == "Description"
+        ret = {
           # rich text label
           "rich_text_title" => _("User Settings"),
           "menu_titles"     => [
             # menu button label
             { "id" => "users--user", "title" => _("&User") },
             # menu button label
-            { "id" => "users--root", "title" => _("&Root Password") }
+            { "id" => "users--root", "title" => _("&Root Password") },
+            # menu button label
+            { "id" => "users--encryption", "title" => _("Password &Encryption Type") }
           ],
           "id"              => "users"
         }
-      elsif @func == "AskUser"
+      elsif func == "AskUser"
         Wizard.OpenAcceptDialog
-        @args = {
+
+        args = {
           "enable_back" => true,
-          "enable_next" => Ops.get_boolean(@param, "has_next", false)
+          "enable_next" => param.fetch("has_next", false),
+          "going_back"  => true
         }
-        @result = :back
-        if Ops.get_string(@param, "chosen_id", "") == "users--root"
+        result = :back
+        case param["chosen_id"]
+        when "users--root"
           UsersSimple.SkipRootPasswordDialog(false) # do not skip now...
-          @result = Convert.to_symbol(
-            WFM.CallFunction("inst_root_first", [@args])
-          )
+          client = "inst_root_first"
+        when "users--user"
+          args["root_dialog_follows"] = false
+          client = "inst_user_first"
         else
-          Ops.set(@args, "root_dialog_follows", false)
-          @result = Convert.to_symbol(
-            WFM.CallFunction("inst_user_first", [@args])
-          )
+          client = "users_encryption_method"
         end
+        result = WFM.CallFunction(client, [args])
 
         Wizard.CloseDialog
 
-        @ret = { "workflow_sequence" => @result }
+        ret = { "workflow_sequence" => result }
         Builtins.y2debug(
           "Returning from users_proposal AskUser() with: %1",
-          @ret
+          ret
         )
       end
-      deep_copy(@ret)
+      ret
```

Comment:
if you touch this file I really recommend to use ProposalClient as base class - https://github.com/yast/yast-yast2/blob/master/library/general/src/lib/installation/proposal_client.rb